dev/NavigationView/NavigationView.h

```diff
@@ -357,6 +372,7 @@ class NavigationView :
     TopNavigationViewDataProvider m_topDataProvider{ this };
 
     winrt::SelectionModel m_selectionModel{};
+    winrt::SelectionModel m_footerSelectionModel{};
```

Comment:
are items in the footer considered separately for selection compared to the rest ? Should they be in the same selection model ? I don't think you can have two selections at the same time, so having two selection models might not be correct.

Reply:
> Would it be simpler to merge the FooterItems and menu items into a single source for selection model ?

Yes, I believe so. Ideally all items, whether in the main list or in the footer list, should have at most one item selected at any given time. The items are navigation destinations, and it's not really possible to navigate to more than one destination page at a time.

Reply:
Ok. I'll move code to single selection model. This should make code simpler

Reply:
Hm. I looked through code again. `SelectionModel` has `Source` property. For now this is `MenuItemsSource` or `MenuItems` for main menu items.

For `FooterMenuItems` i had to create collection and dynamically add or remove `SettingsItem` to the end of it. 

So should i create some array of array as `Source` for `SelectionModel` or just have two selection models synchronized for keep single selection behavior (just like this code do now)?

Reply:
@ojhad any ideas. I think a type that takes two collections and exposes itself as one larger collection would be useful here (just exposure, without copying the items from the individual source collections). That way you can feed MenuItems and FooterMenuItems and expose it as a single merged collection. It can be a completely decoupled implementation that can be useful in other scenarios as well. Thoughts ?

Reply:
Yeah, having larger facade collection sounds like the way to go. I currently don't see it causing any problems with the current NavigationView SelectionModel implementation.

@ad1Dima what is the plan with the footer list for top nav? When the Primary list grows larger than its available size, the items are placed in an overflow menu. That all is handled by a top nav data provider which takes the MenuItems/MenuItemsSource as input.
Is there a similar plan here?

Reply:
>  Is there a way we are handling a large footer list?

For now, it just grow till it can. And it can take all available space in both left and top mode.
For me, bottomMenuItems is purposed for 2 or 3 items like in Groove or MS Store. And may be it should be app developer responsibility to deal with a lot of items on small screen.

Reply:
While app developers do need to keep general layout in mind, NavigationView needs to gracefully support scrolling or overflow. On Windows, a user can resize their window at any time, so no one can guarantee that all items will always fit. This is the reason the primary nav list scrolls in left mode and have overflow in top mode. The main and bottom lists may end up visually merged depending on both the number of items and also window size.

Reply:
For left mode i can try to make single scroll experience for whole pane (with primary nav and footer nav). It will look nice. I did this once.

For top nav i'm not sure what is the best approach. Two overflow menus will look odd

Reply:
@YuliKl 
That's what i did for left menu, is it ok:
![VU9FZ2UQDN](https://user-images.githubusercontent.com/3244664/76926416-e2938a00-690e-11ea-985e-f9194978d2ab.gif)
But i have no idea what to do with Top menu. For now it can be clipped even without footer items. just if menu footer will be too wide:
![image](https://user-images.githubusercontent.com/3244664/76926564-49b13e80-690f-11ea-8087-4b69b1268fca.png)

Reply:
Let's not block this PR on getting bottom items to overflow correctly when in Top mode. I think it's fine to create a separate issue to track this limitation and make a note in the docs.